api/src/main/java/org/apache/iceberg/encryption/NativeFileDecryptParameters.java

```diff
@@ -0,0 +1,100 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.encryption;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Barebone decryption parameters, one object per content file.
+ * Carries the file and column encryption keys (and optional AAD prefix).
+ */
+public class NativeFileDecryptParameters {
+  private ByteBuffer fileAadPrefix;
+  private Map<String, ByteBuffer> columnKeys;
+  private ByteBuffer fileKey;
+
+  private NativeFileDecryptParameters(Map<String, ByteBuffer> columnKeys, ByteBuffer fileKey,
+                                      ByteBuffer fileAadPrefix) {
+    Preconditions.checkState((columnKeys != null && columnKeys.size() > 0) || fileKey != null,
+            "No file or column keys are supplied");
+    this.columnKeys = columnKeys;
+    this.fileKey = fileKey;
+    this.fileAadPrefix = fileAadPrefix;
+  }
+
+  /**
+   * Creates the builder.
+   * @param fileKey per-file encryption key. For example, used as "footer key" DEK in Parquet encryption.
+   */
+  public static Builder create(ByteBuffer fileKey) {
+    return new Builder(fileKey);
+  }
+
+  public static class Builder {
+    private ByteBuffer fileAadPrefix;
+    private Map<String, ByteBuffer> columnKeys;
+    private ByteBuffer fileKey;
+
+    private Builder(ByteBuffer fileKey) {
+      this.fileKey = fileKey;
+    }
+
+    /**
+     * Set column keys.
+     * Key retrieval can require authorization checks and involve interaction with a KMS. In this case, supply
+     * keys only for projected columns.
+     * @param columnKeyMap Map of column names to column keys. Column names must be the original names,
+     *                     used during content file creation. For example, Parquet will use them to find and
+     *                     decrypt the relevant columns.
+     */
+    public Builder columnKeys(Map<String, ByteBuffer> columnKeyMap) {
+      this.columnKeys = columnKeyMap;
+      return this;
+    }
+
+    /**
+     * Set optional AAD prefix.
+     * @param aadPrefix Will be used by Parquet for cryptographic verification of file identity
+     */
+    public Builder aadPrefix(ByteBuffer aadPrefix) {
+      this.fileAadPrefix = aadPrefix;
+      return this;
+    }
+
+    public NativeFileDecryptParameters build() {
+      return new NativeFileDecryptParameters(columnKeys, fileKey, fileAadPrefix);
+    }
+  }
+
+  public ByteBuffer aadPrefix() {
```

Comment:
Should we name it as getAadPrefix()?

Reply:
This class is a bundle of parameters, so `get` is implicit.